QuizApp/Screens/HomePage/HomePageViewModel.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  HomePageViewModel.swift
+//  QuizApp
+//
+//  Created by Ana on 12/16/24.
+//
+
+import Foundation
+
+final class HomePageViewModel {
+    private let dataCommunication = DataCommunication.shared
+    private var currentUser: UserDataModel?
```

Comment:
The currentUser variable is declared as optional, but the HomePageViewModel appears to rely heavily on it for its functionality. Since the home page cannot logically exist without a currentUser, making it optional is unnecessary and introduces redundant checks.